### DISCUSSION TITLE
Please allow users to specify compiler

### DISCUSSION THREAD

**Discussion by byxrolland** [microsoft/vcpkg#19976] [Ideas]
****

**Is your feature request related to a problem? Please describe.**
On Linux, there exist multiple shells and versions of compilers. But vcpkg always detect those by itself. For example, I have gcc-11 installed by linuxbrew but vcpkg's bootstrap.sh always use the /bin/sh shell which is not the user login shell and whose environment variables only specify the system presinstalled gcc (gcc-9).

Even if I can use
```bash
$ bash ./bootstrap.sh
```
to overcome this, vcpkg still chooses an compiler for me when installing and upgrading packages. This is not desirable.

**Proposed solution**
 
It would be nice if we can set the default compiler using

```bash
$ vcpkg --set_default_cc=gcc-11
```

and specify compilers when installing and upgrading packages with something like
```bash
$ vcpkg install --cc=gcc-10 eigen3

$ vcpkg upgrade --cc=gcc=clang boost --no-dry-run
```

**Additional context**
It would also be great if there are options that allow user to use their own MSYS2 and cmake in order to save space.

**Comment by Osyotr** [comment]
It is already possible:
https://github.com/microsoft/vcpkg/blob/master/docs/users/triplets.md#vcpkg_chainload_toolchain_file

**Reply by byxrolland** [reply]
Possible but too cumbersome for only specifying the compiler. It should be achieved by simple command line options, just like homebrew.
